src/current/advisories/a99561.md

```diff
@@ -0,0 +1,37 @@
+
+In CockroachDB v22.2 and later, the implementation for [`ALTER TABLE`](../v22.2/alter-table.html#add-column) began using the new [declarative schema changer](../v23.1/online-schema-changes.html#declarative-schema-changer) framework. When adding a new column using the new declarative schema changer, CockroachDB retrieves a unique secondary index mistakenly, assuming it to be a primary index. As a result, the secondary index is modified to store the new columns. CockroachDB expects there to be only one primary index, but in the case outlined here, there are two, because a secondary index has been incorrectly labeled as a primary one. This will cause the new column data to be present only in the secondary index and could lead to irrecoverable data loss, if the secondary index is deleted by the user.
+
+## Statement
+
+This is resolved in CockroachDB by [#105828](https://github.com/cockroachdb/cockroach/pull/105828), which makes CockroachDB automatically repair the encoding type for secondary indexes when accessing the table.
+
+The fix will be applied to v22.2.12 and v23.1.6.
+
+This public issue is tracked by [#99561](https://github.com/cockroachdb/cockroach/issues/99561).
+
+## Mitigation
+
+Users of CockroachDB v22.2.x are encouraged to upgrade to [v22.2.12](../releases/v22.2.html) upon its release. Similarly, users of v23.1.0 — v23.1.5 are encouraged to update to [v23.1.6](../releases/v23.1.html) upon its release.
+
+Until such an upgrade, run the query outlined in [this gist](https://gist.github.com/fqazi/2b71b05ce8f6b028267ed3b4498c4f6a) to check if your cluster is in a potentially corrupt state. If the query returns any data, contact our [support team](https://support.cockroachlabs.com/) to resolve the inconsistency. Until our support team has had a chance to fix the cluster, refrain from running any schema changes on the impacted table including [`DROP INDEX`](../v23.1/drop-index.html) or [`ALTER TABLE`](../v23.1/alter-table.html) as it could lead to data loss.
+
+## Impact
+
+Upon upgrade to CockroachDB v22.2.0 and above, a bug from v20.2 and v21.1 could cause the secondary index to be modified upon `ALTER TABLE..ADD COLUMN` instead of the existing primary index. If the user drops the secondary index, all data related to the new column will be deleted resulting in irrecoverable data loss.
```

Comment:
```suggestion
Upon upgrade to CockroachDB v22.2.0 and above, a bug from v20.2 and v21.1 could cause the secondary index to be modified upon `ALTER TABLE..ADD COLUMN` instead of the existing primary index. If the user drops the secondary index, all data related to the new column will be deleted resulting in irrecoverable data loss. Additionally, until this cluster is repaired, there is a risk of inconsistent query results as the primary index does not contain data for the newly added columns.
```